pymc/sampling/mcmc.py

```diff
@@ -1256,12 +1260,18 @@ def _iter_sample(
 
             yield diverging
     except KeyboardInterrupt:
+        if isinstance(trace, ZarrChain):
+            trace.record_sampling_state(step=step)
```

Comment:
Should we make this part of `trace.close`?

Reply:
I don't want to change the signature of `close` and I don't want to make `trace` necessarily have a reference to the `step`. We could add a `record_sampling_state` method to `NDArray` though

Reply:
Sounds good, and doesn't have to be done in this PR

Reply:
Opened #7621 to track this.